Cartesian_kernel/include/CGAL/Cartesian/function_objects.h

```diff
@@ -3031,6 +3031,24 @@ namespace CartesianKernelFunctors {
 
       return construct_vector(vx, vy, vz);
     }
+
+    Vector_3
+    operator()( Origin o , const Point_3& q, const Point_3& r ) const
```

Comment:
It is better to pass it by value, as Andreas did.

Reply:
> Could you use `const Origin& o` here instead of `Origin o` or doesn't it matter?

That would be a pessimization. `Origin` is an empty class. Passing it by reference to const forces the compiler to pass the address of `CGAL::ORIGIN` on the stack, whereas passing it by value can be optimized out.

Reply:
@lrineau I think I already knew this. Typically, pass by reference when the object you are passing is much larger than the size of a pointer. However, I've seen it in other codebases (such as Qt), where it's more a case of consistency than performance. I prefer performance ;)